alicloud/data_source_alicloud_disks_test.go

```diff
@@ -0,0 +1,117 @@
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAlicloudDisksDataSourceBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudDataSourceID("data.alicloud_disks.disks"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.#", "1"),
+					resource.TestCheckResourceAttrSet("data.alicloud_disks.disks", "disks.0.id"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.name", "tf-testAccCheckAlicloudDisksDataSourceBasic"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.description", "tf-testAccCheckAlicloudDisksDataSourceBasic_description"),
+					resource.TestCheckResourceAttrSet("data.alicloud_disks.disks", "disks.0.region_id"),
+					resource.TestCheckResourceAttrSet("data.alicloud_disks.disks", "disks.0.availability_zone"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.status", "Available"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.type", "data"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.category", "cloud_efficiency"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.encrypted", "off"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.size", "20"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.image_id", ""),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.snapshot_id", ""),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.instance_id", ""),
+					resource.TestCheckResourceAttrSet("data.alicloud_disks.disks", "disks.0.creation_time"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.attached_time", ""),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.detached_time", ""),
+					resource.TestCheckResourceAttrSet("data.alicloud_disks.disks", "disks.0.expiration_time"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.tags.%", "2"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.tags.Name", "TerraformTest"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAlicloudDisksDataSource_filterByAllFields(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAlicloudDisksDataSourceFilterByAllFields,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudDataSourceID("data.alicloud_disks.disks"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.#", "1"),
+					resource.TestCheckResourceAttr("data.alicloud_disks.disks", "disks.0.name", "tf-testAccCheckAlicloudDisksDataSourceFilterByAllFields"),
+				),
+			},
+		},
+	})
+}
+
+const testAccCheckAlicloudDisksDataSourceBasic = `
```

Comment:
Can you add more test cases to cover all of parameters?

Reply:
See TestAccAlicloudDisksDataSource_filterByAllFields